### PR TITLE
Medical - Option for treatment to require holstering/lowering weapon.

### PR DESCRIPTION
**Pull request by ddm999**
Adds Medical options (off by default) to disallow treatment while a weapon is active or raised.
--
Based on #654

What this does:
- Adds a "Holster Required" entry to the ACE Medical options (under Treatment).
- - When set to "Holstered only", medical treatments can only be performed if there is no active weapon (either by not having a weapon or by pressing the ACE Weapons > Holster Weapon control).
- - When set to "Lowered or Holstered", medical treatments can only be performed if there is no active weapon, or the weapon is lowered.
- - When set to "Disabled" (default), nothing changes.
- Adds a "Bypass Holster Required for Examination" entry to the ACE Medical options (also under Treatment, off by default).
- - When checked, treatments in the "examine" category are allowed regardless of weapon state and "Holster Required" setting.
- - This allows the Check Pulse, Check Blood Pressure, Response actions to be performed without lowering or holstering.

What this ***doesn't*** do (for potential consideration):
- Hide the medical interaction from limbs that have no available options (so it doesn't really 'de-clutter', other than preventing the treatment options from appearing and disappearing while you're looking around a body with weapon raised).
- - This means you can still see the Injuries and Triage Card display on the left by hovering over limbs, if "Enable Medical Actions" isn't set to "Disabled".
- Wait for any unequip / lower animations to finish before you're allowed to perform medical actions.
- Prevent access to the Medical Menu, even if no treatments are available.
- Prevent access to the Triage Card via Torso.
- Give the player an indication of why they can't perform medical actions if settings are enabled but they have their gun raised / equipped.
- Distinguish between treatments to allow some be allowed when weapon is lowered while others only allowed when no weapon is active.
- Distinguish between self-treatment and treating others.

Noted potential "issues":
- when raising weapon and opening the Medical Menu quickly, there's a delay before some treatments disappear (note: they can't/won't be performed if clicked)
- at a glance, "Holster Required: Holstered only" doesn't really make much sense. Might need to change the DisplayName.

Also, I realise that the request this is based on is very, very old, and I'd imagine some of the reasoning behind why this was wanted no longer applies.
I still made this PR because it seemed relatively simple for someone with little experience of the project. :)

### CODE REVIEW

**Comment by ddm999** [comment]
Just noticed this doesn't play nice with vehicles (keeps your state when you entered it), changed title to WIP and am going to look into it tomorrow.

**Comment by ddm999** [comment]
Fixed the issue with vehicles (holster setting is basically ignored when either medic or patient is in a vehicle: so pointing a gun at a vehicle while you're healing someone inside is allowed).
If anyone's interested in any of the stuff I mentioned in the PR description (under "what this (currently) doesn't do") - let me know, and specify what you'd personally prefer. I'll see if I can add it to this PR as another non-default option. :)

Otherwise, this small optional feature is ready, and I'll remove the WIP tag if there's no extra features wanted.

File: addons/medical_treatment/functions/fnc_canTreat.sqf [hunk 1]
```diff
@@ -27,6 +27,14 @@ isClass _config
 && {_medic != _patient || {GET_NUMBER_ENTRY(_config >> "allowSelfTreatment") == 1}}
 && {[_medic, GET_NUMBER_ENTRY(_config >> "medicRequired")] call FUNC(isMedic)}
 && {
+    // Handle holster settings
+    GVAR(holsterRequired) == 0
+    || !(vehicle _medic isEqualTo _medic) // medic is in a vehicle, so weapon is considered holstered
+    || !(vehicle _patient isEqualTo _patient) // patient is in a vehicle, ^
```
Comment (dedmen) [inline addons/medical_treatment/functions/fnc_canTreat.sqf]: A simple != might be faster than a ! and isEqualTo.

Comment (ddm999) [reply addons/medical_treatment/functions/fnc_canTreat.sqf]: Ah ok, from what I'd read on the wiki it sounded like isEqualTo might've been faster. Will correct now.

**Comment by bux** [comment]
One of the best PR descriptions of a first time contributor. Cheers. <3

File: addons/medical_treatment/functions/fnc_canTreat.sqf (hunk 1 above)
Comment (dedmen) [reply addons/medical_treatment/functions/fnc_canTreat.sqf]: isEqualTo is a TINY bit faster than ==.
But ! (x isEqualTo y) are more script commands and is slower.

File: addons/medical_treatment/functions/fnc_canTreat.sqf [hunk 2]
```diff
@@ -27,13 +27,13 @@ isClass _config
...
 && {
-    // Handle holster settings
+    // Handle holster settings [disabled, lowered, loweredExam, holster, holsterExam]
     GVAR(holsterRequired) == 0
-    || vehicle _medic != _medic // medic is in a vehicle, so weapon is considered holstered
-    || vehicle _patient != _patient // patient is in a vehicle, ^
-    || {GVAR(holsterNoReqExamine) && getText (_config >> "category") == "examine"} // if examine bypass is on
-    || currentWeapon _medic isEqualTo "" // weapon is holstered
-    || {GVAR(holsterRequired) == 1 && weaponLowered _medic} // if just lowered is allowed
+    || {vehicle _medic != _medic} // medic is in a vehicle, so weapon is considered holstered
```
Comment (jonpas) [inline addons/medical_treatment/functions/fnc_canTreat.sqf]: I would separate all those conditions into a separate function, for clarity. They are all here for holstering only.

**Comment by Max255PL** [comment]
We were testing this PR lately and found an issue with healing while prone.
It still recognizes the weapon as "up" and you can't perform any medical actions. 
Yes, you can put it on the back but sometimes it's a suboptimal solution.

A good solution for this would be to add a setting to check for weapon safety on, be it with combination with other settings or by itself.

**Comment by PabstMirror** [comment]
If you are prone, your weapon is always lowered 😄

**Pull request merged**